metals/src/main/scala/scala/meta/internal/metals/FileWatcher.scala

```diff
@@ -129,22 +123,14 @@ final class FileWatcher(
       // this error won't affect correctness of Metals.
       case _: ju.ConcurrentModificationException =>
     }
-    activeFileWatcher.foreach(_.close())
-    fileWatching.cancel(false)
     directoryWatching.cancel(false)
   }
 
   class DirectoryListener extends DirectoryChangeListener {
     override def onEvent(event: DirectoryChangeEvent): Unit = {
-      if (!Files.isDirectory(event.path())) {
+      if (event.eventType() == EventType.OVERFLOW) {
```

Comment:
we could add a comment why it has dedicated branch that is doing the same logic.

Reply:
added